ostree-upload-server.py

```diff
@@ -128,9 +139,33 @@ def upload(self):
                 os.close(f)
                 upload.save(real_name)
 
-                task = ReceiveTask(upload.filename, real_name,
-                                   self._repo)
+                repo_name = request.form.get('target-repo', None)
+                logging.info("Target repo: {}".format(repo_name))
+
+                if (not repo_name) and self._enforce_target_repo:
+                    logging.error("Target repo enforcement set but target repo not provided!")
+                    return self._response(400, "ERROR! target-repo parameter not set!")
+
+                # Verify that this param is not trying anything funky
+                if not UploadWebApp.REPO_NAME_PATTERN.match(repo_name):
+                    logging.error("Target repo name ({}) is not valid!".format(repo_name))
+                    return self._response(400, "ERROR! target-repo parameter is not valid!")
```

Comment:
I really don't want to be allowing arbitrary repos to be created from a remote caller. Please make the configuration of the repositories explicit. What I suggest is something like how the remotes are handles in ostree-upload-server.conf. Like:
```
[repo-foo]
path = /srv/repos/foo

[repo-bar]
path = /srv/repos/bar
```
Then make a dict with the `repo-` stripped from the section name or whatever.

After that, no command line arg is needed and the server should bail early if there are no defined repos.